[ bug ]登录报错

在提交 issue 前，请查阅以下资源，请先进行搜索来保证没有类似的 issue。
[文档](https://github.com/hhyo/archery/wiki) | [FAQ](https://github.com/hhyo/archery/wiki/FAQ) 

### 重现步骤
v1.8.3升级到v1.8.4



### 错误日志
启动服务
[2022-05-05 10:51:03 +0800] [43] [INFO] Starting gunicorn 20.0.4
[2022-05-05 10:51:03 +0800] [43] [INFO] Listening at: http://127.0.0.1:8888 (43)
[2022-05-05 10:51:03 +0800] [43] [INFO] Using worker: sync
[2022-05-05 10:51:03 +0800] [44] [INFO] Booting worker with pid: 44
[2022-05-05 10:51:03 +0800] [45] [INFO] Booting worker with pid: 45
[2022-05-05 10:51:04 +0800] [46] [INFO] Booting worker with pid: 46
[2022-05-05 10:51:04 +0800] [47] [INFO] Booting worker with pid: 47
/opt/venv4archery/lib/python3.9/site-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
[2022-05-05 10:51:22 +0800] [47] [ERROR] Error handling request /login/
Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/base.py", line 167, in _get_response
    callback, callback_args, callback_kwargs = self.resolve_request(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/base.py", line 290, in resolve_request
    resolver_match = resolver.resolve(request.path_info)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 549, in resolve
    for pattern in self.url_patterns:
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 593, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
    response = response or self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv4archery/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 134, in handle
    self.handle_request(listener, req, client, addr)
  File "/opt/venv4archery/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/wsgi.py", line 133, in __call__
    response = self.get_response(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/base.py", line 130, in get_response
    response = self._middleware_chain(request)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/opt/venv4archery/lib/python3.9/site-packages/django/core/handlers/exception.py", line 141, in handle_uncaught_exception
    callback, param_dict = resolver.resolve_error_handler(500)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 606, in resolve_error_handler
    callback = getattr(self.urlconf_module, 'handler%s' % view_type, None)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/resolvers.py", line 586, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/archery/urls.py", line 7, in <module>
    path('api/', include(('sql_api.urls', 'sql_api'), namespace="sql_api")),
  File "/opt/venv4archery/lib/python3.9/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/python3/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/archery/sql_api/urls.py", line 7, in <module>
    from . import api_user, api_instance, api_workflow
  File "/opt/archery/sql_api/api_user.py", line 6, in <module>
    from .pagination import CustomizedPagination
  File "/opt/archery/sql_api/pagination.py", line 7, in <module>
    class CustomizedPagination(PageNumberPagination):
  File "/opt/archery/sql_api/pagination.py", line 11, in CustomizedPagination
    page_size = settings.REST_FRAMEWORK['PAGE_SIZE'] if 'PAGE_SIZE' in settings.REST_FRAMEWORK.keys() else 20
  File "/opt/venv4archery/lib/python3.9/site-packages/django/conf/__init__.py", line 83, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'REST_FRAMEWORK'


### 版本信息
应用版本/分支：v1.8.4

部署方式：Docker

--- a/src/docker-compose/archery/settings.py
+++ b/src/docker-compose/archery/settings.py
@@ -3,6 +3,7 @@
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 import os
+from datetime import timedelta
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
@@ -32,6 +33,9 @@ INSTALLED_APPS = (
     'sql',
     'sql_api',
     'common',
+    'rest_framework',
+    'django_filters',
+    'drf_spectacular',
 )
 
 MIDDLEWARE = (
@@ -169,6 +173,49 @@ CACHES = {
             "PASSWORD": "123456"
         }
     }
+}
+
+# API Framework
+REST_FRAMEWORK = {
+    'DEFAULT_SCHEMA_CLASS': 'drf_spectacular.openapi.AutoSchema',
+    'DEFAULT_RENDERER_CLASSES': ('rest_framework.renderers.JSONRenderer',),
+    # 鉴权
+    'DEFAULT_AUTHENTICATION_CLASSES': (
+        'rest_framework_simplejwt.authentication.JWTAuthentication',
+        'rest_framework.authentication.SessionAuthentication',
+    ),
+    # 权限
+    'DEFAULT_PERMISSION_CLASSES': ('sql_api.permissions.IsInUserWhitelist',),
+    # 限速（anon：未认证用户  user：认证用户）
+    'DEFAULT_THROTTLE_CLASSES': (
+        'rest_framework.throttling.AnonRateThrottle',
+        'rest_framework.throttling.UserRateThrottle',
+    ),
+    'DEFAULT_THROTTLE_RATES': {
+        'anon': '120/min',
+        'user': '600/min'
+    },
+    # 过滤
+    'DEFAULT_FILTER_BACKENDS': ('django_filters.rest_framework.DjangoFilterBackend',),
+    # 分页
+    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
+    'PAGE_SIZE': 5,
+}
+
+# Swagger UI
+SPECTACULAR_SETTINGS = {
+    'TITLE': 'Archery API',
+    'DESCRIPTION': 'OpenAPI 3.0',
+    'VERSION': '1.0.0',
+}
+
+# API Authentication
+SIMPLE_JWT = {
+    'ACCESS_TOKEN_LIFETIME': timedelta(hours=4),
+    'REFRESH_TOKEN_LIFETIME': timedelta(days=3),
+    'ALGORITHM': 'HS256',
+    'SIGNING_KEY': SECRET_KEY,
+    'AUTH_HEADER_TYPES': ('Bearer',),
 }
 
 # LDAP

